core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -143,6 +146,9 @@ public void start(ClientStreamListener listener) {
         }
       };
     }
+    headers.put(
+        GrpcUtil.USER_AGENT_KEY,
+        GrpcUtil.getGrpcUserAgent("inprocess", userAgent));
```

Comment:
nit: the agent should be a final var for this instance.

Reply:
I'll do that as a follow-up.